### PR TITLE
Implement getPointTrespass Function for Navigator

### LINKED ISSUE #355: Implement getPointTrespass
### Description of the task

Implement a `getPointTrespass` function that returns how much a point is trespassing in another point's space. 

### Outline the task in more detail

This can be thought of as a generalization of the `get_ball_...._trespass` and `get_robot_..._trespass` functions, since all they are is really points. This is really just returning the distance between the point and the trespass point if the point is within a threshold of the trespass point, and should return 0 otherwise. Look at the old functions for examples.

Eg. `getPointTrespass(Point p1, Point p2, double tresoass_threshold)`

The old code is in the GitLab repo in `software/ai/navigator/util.cpp`

### Acceptance criteria

- [ ] Implement the `getPointTrespass` function
- [ ] Unit test the function

### PR DESCRIPTION
**Pull request by jessicahe21**
### Description
Implemented `getPointTrespass` function under `navigator/util.*`. The purpose of this function is to generalize the `get_ball/robot_..._trespass` functions from the [old code](https://gitlab.com/thunderbotsfc/Software-Firmware/blob/master/software/ai/navigator/util.cpp) as they all use points. 

This function returns the distance that a given point has trespassed within the given threshold of the trespass point. Otherwise, it will return 0.

### Testing Done
Unit tests are written for 4 scenarios: within, within but closer, equal, or greater than the given threshold. 
All tests pass.

### Resolved Issues
resolves #355 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Comment by jessicahe21** [comment]
In issue #355 it was mentioned that,
> This can be thought of as a generalization of the `get_ball_...._trespass` and `get_robot_..._trespass` functions ...

In the old code there is a myriad of implementations for `get_..._trespass`. To my understanding, in order to retain the old implementations, some sort of `Ball`, `Player`, `Field` etc. input is needed.

I'm afraid my current implementation is a little too over simplified as opposed to those in the old code. 
 Should the function be responsible for taking care of individual cases? (e.g. `get_ball_stop_trespass`, `ball_tiny`, `ball_regular`, etc)

File: src/thunderbots/software/ai/navigator/util.h [hunk 1]
```diff
@@ -18,3 +18,17 @@
  */
 double calculateTransitionSpeedBetweenSegments(const Point &p1, const Point &p2,
                                                const Point &p3, double final_speed);
+
+/**
+ * Calculates how much a point is trespassing in another point's space.
+ *
+ * Returns the distance between the given point and the threshold point if it is within
```
Comment (MathewMacDougall) [inline src/thunderbots/software/ai/navigator/util.h]: The return is slightly backwards. We want to return how far inside the "trespass boundary" the point is from the other, so the trespass value is 0 outside the boundary, and increases as the point approaches the other. Should be an easy fix.

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
One quick comment, otherwise looks good

File: src/thunderbots/software/ai/navigator/util.h (hunk 1 above)
Comment (jessicahe21) [reply src/thunderbots/software/ai/navigator/util.h]: I see! Here is my current interpretation for what the trespass value/distance is:
untitled (5)
Is this understanding correct?

Currently for the edge case where the trespass value is equal to the threshold, 0 is returned.
 
As well, I for the life of me couldn't quite figure out why `dist(p1, p2)` returns an "undefined reference" error in line 14 of `util.cpp` so I used `(p1 - p2).len()` instead for now.

**Comment by MathewMacDougall** [comment]
Sorry I didn't see your previous comment regarding the "generalization".

We are trying to avoid remaking all the "trespass" functions that took a `Ball`, `Robot`, etc. because all they are really doing is a `Point Trespass`. To re-create this old functionality you can just call the `PointTrespass` and provide the ball position for example.

`double trespass = getPointTrespass(ball.position(), some_point_to_avoid);`

File: src/thunderbots/software/ai/navigator/util.h (hunk 1 above)
Comment (MathewMacDougall) [reply src/thunderbots/software/ai/navigator/util.h]: Yes your understanding is correct (nice diagram btw).

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Pull request merged**